### ISSUE TITLE
Clj-kondo warns that reify method type hints are unused 

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1588]
**version**
v2022.02.09

**platform**
macOS 10.15.7
jvm and binary

**problem**
The [`reify`](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/reify) macro allows for type hints to be specified on methods.
> The return type can be indicated by a type hint on the method name

(Sometimes they are even necessary!)

For a totally contrived example:
```Clojure
(ns fiddle
  (:import (java.lang.management PlatformManagedObject)
           (javax.management ObjectName)))

(reify PlatformManagedObject
  (^ObjectName getObjectName [_this]))
```
Ping me for something more real if you need it.

**repro**
```Shell
$ clj-kondo --lint fiddle.clj                                                                                        
fiddle.clj:3:30: warning: Unused import ObjectName
linting took 12ms, errors: 0, warnings: 1
```

**expected behavior**
I expect that clj-kondo recognize that `ObjectName` is used and not issue a warning.

**work-around**
A simple work-around is to suppress the warning with a `#_{:clj-kondo/ignore [:unused-import]}` like so:
```Clojure
(ns fiddle
  (:import (java.lang.management PlatformManagedObject)
           #_{:clj-kondo/ignore [:unused-import]}
           (javax.management ObjectName)))

(reify PlatformManagedObject
  (^ObjectName getObjectName [_this]))
```
